### DISCUSSION TITLE
New fonts and non-ASCII characters

### DISCUSSION THREAD

**Discussion by katze** [abey79/vpype#497] [Q&A]
Hello,
I just discovered this great project while looking for a way to script the creation of svg from dynamic texts.

I saw that there were already a lot of fonts: https://github.com/abey79/vpype/tree/master/vpype/fonts but none with all the characters used in French like: éêèçàùî

How can I add these characters? Besides, what is this pickle format?

**Comment by abey79** [comment]
Hi @katze.

The current font situation in vpype is sub-optimal. What i've done so far is "stealing" the [fonts](https://github.com/fogleman/axi/blob/master/axi/hershey_fonts.py) from the [axi](https://github.com/fogleman/axi) project and pickle them instead of storing them in Python source format ([pickle](https://docs.python.org/3/library/pickle.html) is a python-native binary format). This is neither standard, nor easily extensible. These fonts (or similar ones) exists in SVG format (e.g.[here](https://gitlab.com/oskay/hershey-text/-/tree/master/hershey-text/svg_fonts)). Adding i18l glyphs would be much easier with these.

So ideally the first step would be to add support for SVG fonts to vpype, which would then enable using custom fonts with the needed glyphs.

**Comment by golanlevin** [comment]
Adding my +1 for SVG 1.1 fonts in vpype.

Of possible interest, I maintain a very large archive of single-stroke fonts here: 
https://github.com/golanlevin/p5-single-line-font-resources/blob/main/README.md
Naturally this includes all of the Hershey and EMSL SVG fonts, and many others. 

Two carefully-designed single-stroke fonts of particular relevance to this discussion are [Relief](https://github.com/isdat-type/Relief-SingleLine/), developed by François Chastanet at the Institut Supérieur des Arts et du Design de Toulouse, and [ISO 3098](https://www.southype.com/Commerce/iso-3098-standard-an-in-depth-guide-to-labeling-and-lettering/), an international standard for lettering in technical drawings. Both have extended character sets for European languages, and are available in SVG 1.1 font format [in my repository here](
https://github.com/golanlevin/p5-single-line-font-resources/blob/main/README.md#p5js-parserrenderer-for-single-line-svg-fonts).

best.
Golan
